### ISSUE TITLE
Broadcasting error when forecasting with `method="simulation"` and multiple forecasting periods.

### ISSUE THREAD

**Issue by ZappD0S** [bashtage/arch#551]
I'm having trouble doing forecasts when using exogenous regressors. When using `method="simulation"` and having a number of forecasting periods greater than 1 the `forecast` method fails, raising the following exception:

> ValueError: operands could not be broadcast together with shapes (1000,) (3,) (1000,)

I also tried with the code that I found in [this notebook](https://arch.readthedocs.io/en/latest/univariate/univariate_forecasting_with_exogenous_variables.html) from the `arch` docs. At first it seems to be fine because in the notebook the number of forecasting periods is 1. But if the `start` parameter is specified, the error appears.

The documentation of the `forecast` method says that the exogenous regressors can be specified with a dict where each value is a 2-d array with shape (nforecast, horizon) or (nobs, horizon). I tried both shapes and the error is present in both cases. Here is the snippet that I'm using:

```python
import numpy as np
import pandas as pd
from arch.__future__ import reindexing
from arch.univariate import ARX, GARCH, ZeroMean, arch_model

burn = 250

x_mod = ARX(None, lags=1)
x0 = x_mod.simulate([1, 0.8, 1], nobs=1000 + burn).data
x1 = x_mod.simulate([2.5, 0.5, 1], nobs=1000 + burn).data

resid_mod = ZeroMean(volatility=GARCH())
resids = resid_mod.simulate([0.1, 0.1, 0.8], nobs=1000 + burn).data

phi1 = 0.7
phi0 = 3
y = 10 + resids.copy()
for i in range(1, y.shape[0]):
    y[i] = phi0 + phi1 * y[i - 1] + 2 * x0[i] - 2 * x1[i] + resids[i]

x0 = x0.iloc[-1000:]
x1 = x1.iloc[-1000:]
y = y.iloc[-1000:]
y.index = x0.index = x1.index = np.arange(1000)

x0_oos = np.empty((1000, 10))
x1_oos = np.empty((1000, 10))
for i in range(10):
    if i == 0:
        last = x0
    else:
        last = x0_oos[:, i - 1]
    x0_oos[:, i] = 1 + 0.8 * last
    if i == 0:
        last = x1
    else:
        last = x1_oos[:, i - 1]
    x1_oos[:, i] = 2.5 + 0.5 * last

exog = pd.DataFrame({"x0": x0, "x1": x1})
mod = arch_model(y, x=exog, mean="ARX", lags=1)
res = mod.fit(disp="off")

nforecast = 3

# DECOMMENT ONE OF THE FOLLOWING LINES
# exog_fcast = {"x0": x0_oos[-nforecast:], "x1": x1_oos[-nforecast:]} # case with shape (nforecast, horizon)
# exog_fcast = {"x0": x0_oos, "x1": x1_oos} # case with shape (nobs, horizon)

forecasts = res.forecast(
    horizon=10, x=exog_fcast, start=1000 - nforecast, method="simulation"
)
```
This code is mostly the same as the original. The only part that I changed at the end. In the original the `exog_fcast` was built in this way:
```python
exog_fcast = {"x0": x0_oos[-1:], "x1": x1_oos[-1:]}
```
and also I assigned the `start` and `method` parameters in the `forecast` method call. I'm using the latest version of `arch` (5.1.0).

**Comment by bashtage** [comment]
Clearly a bug.  Need to determine what shapes things should be here.

**Comment by bashtage** [comment]
Should be fixed in #552.
